File: go/board.go
package main

type Board struct {
	Squares []Square
}

type Square struct {
	xPos  int
	yPos  int
	Piece *Piece
}

type Piece struct {
	Type   string
	Colour string
}

func NewBoard() Board {
	return Board{
		Squares: []Square{{xPos: 1, yPos: 1, Piece: &Piece{Type: "rook", Colour: "white"}}},
	}
}

File: rust/src/board.rs
use crate::piece::Colour;

use crate::piece::Piece;

pub struct Board {
    pub squares: Vec<Square>,
}

impl Board {
    pub fn new() -> Board {
        Board {
            squares: vec![Square {
                pos_x: 1,
                pos_y: 1,
                piece: Some(Piece::new_rook(Colour::White)),
            }],
        }
    }
}

pub struct Square {
    pub pos_x: u8,
    pub pos_y: u8,
    pub piece: Option<Piece>,
}
